src/test/java/com/puppycrawl/tools/checkstyle/utils/UnmodifiableCollectionUtilTest.java

```diff
@@ -0,0 +1,51 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import static com.puppycrawl.tools.checkstyle.checks.coding.EqualsAvoidNullCheck.MSG_EQUALS_AVOID_NULL;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.EqualsAvoidNullCheck;
+
+public class UnmodifiableCollectionUtilTest extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/utils/unmodifiablecollectionutil";
+    }
+
+    @Test
+    public void testFullIdentCreateFullIdentBelow() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(EqualsAvoidNullCheck.class);
```

Comment:
what blocs us to move this test to EqualsAvoidNullCheckTest class ?

We simply need extension of existing tests in pitest-coding-1 group.

Reply:
` Collections.unmodifiableSet(methodCalls);` the code is mutated as methodCalls.
The methodcalls is been not modifying so for Collections.unmodifiableSet(methodCalls); it is difficult to find test case. with pure ut we can achieve it. as we done in https://github.com/checkstyle/checkstyle/commit/d4ccb63641e4c89348b47161000f0c792fdaf3f3

but to avoid pure ut we create a class to wrap this method so situation is same for here not finding test for kill this mutation. this only cover coverage not mutation thats why suppressed from pitest-coding-1 and pitest-util profile

Reply:
Please move test that you wrote to EqualsAvoidNullCheckTest.
It should be just another test of this Check.

Reply:
https://github.com/checkstyle/checkstyle/pull/13127#discussion_r1234577227  --> Done

Reply:
Your changes can not affect link validation, we can ignore it for Pitest activity

Reply:
done

Reply:
I was wrong, and code changes can affect linkcheck validation, example https://github.com/checkstyle/checkstyle/commit/852339d1e8cde08045a31e4beb7e281bbb246e4a
We need to be accurate.

I still confused, why test method is not in Check test class.

Reply:
> I still confused, why test method is not in Check test class.

If you are asking from this perspective https://github.com/checkstyle/checkstyle/pull/13127#discussion_r1234577227 I have made changes.

or The reason to create this class and create this test is only one to avoid the Collections.unmodifiableSet(); mutation
this test is created only to get code coverage If I don't create this test the coverage would be 0.

and to avoid the pure unit testing I have created this test in which the check is able to cover the code.

apart of this if you have any confusion let me know

Reply:
I am sorry. I don't know what misconce[tuion happen with me so i added this test case. I am going to remove them.

Sorry for the time waste